Subject: Control output of header in crontab -e
Last-Update: 2010-04-11

Change the default behaviour of outputting a warning header to only outputting
it when specifically requested so by setting CRONTAB_NOHEADER=N.

Acked-by: Christian Kastner <debian@kvr.at>
Index: patched/crontab.c
===================================================================
--- patched.orig/crontab.c	2010-05-06 18:18:52.595435667 +0200
+++ patched/crontab.c	2010-05-06 18:18:56.391036364 +0200
@@ -292,6 +292,10 @@
 	char	n[MAX_FNAME];
 	FILE	*f;
 	int	ch;
+#ifdef DEBIAN
+	int     x;
+	char    *ctnh;
+#endif
 
 	log_it(RealUser, Pid, "LIST", User);
 	(void) snprintf(n, MAX_FNAME, CRON_TAB(User));
@@ -307,6 +311,30 @@
 	/* file is open. copy to stdout, close.
 	 */
 	Set_LineNum(1)
+#ifdef DEBIAN
+	  /* DEBIAN: Don't list header lines unless CRONTAB_NOHEADER is
+	     'N'. */
+	  /* ignore the top few comments since we probably put them there.
+	   */
+	  if (!(ctnh = getenv("CRONTAB_NOHEADER")) ||
+	      toupper(*ctnh) != 'N') 
+	    {
+	    for (x = 0;  x < NHEADER_LINES;  x++) {
+	      ch = get_char(f);
+	      if (EOF == ch)
+		break;
+	      if ('#' != ch) {
+		putchar(ch);
+		break;
+	      }
+	      while (EOF != (ch = get_char(f)))
+		if (ch == '\n')
+		  break;
+	      if (EOF == ch)
+		break;
+	    }
+	  }
+#endif
 	while (EOF != (ch = get_char(f)))
 		putchar(ch);
 	fclose(f);
